research/conditional/utils/conditional_trainer.py

```diff
@@ -134,6 +137,25 @@ def _after_step_operations(self, step):
         self.model.forward_pass_cache.clear()
         self.layer_manager.manage_learnable_temperature(step)
 
+    def _final_eval(
+        self,
+        n_steps: int,
+    ):
+        if self.current_step == n_steps:
+            batches = [
+                self.final_eval_dataloader.get_batch()
+                for _ in range(
+                    self.n_sequences_final_eval // self.final_eval_dataloader_batch_size
+                )
+            ]
+
+            self._eval_single_variant(
```

Comment:
It seems like the function `_final_eval`() admits n_steps as argument, for the option of possibly making eval in the middle of the training (if `n_steps` arument `!= args.n_steps`).` If so, `varinat_name` an `stage` arguments schould be different. I suggest either adding them as arguments to `_final_eval` or deleting `n_steps` argument and taking it from `args`.

Reply:
We do not pass `args` to this funciton. `n_steps` is only for check if it is time for final eval